src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -91,6 +94,42 @@ def get_registry_by_name(cli_ctx, registry_name, resource_group_name=None):
     return client.get(resource_group_name, registry_name), resource_group_name
 
 
+def get_registry_by_registry_name(cli_ctx, registry_name):
+    """Returns a Registry object for the specified registry name.
+    "param str registry_name: The name of the container registry.
+    """
+    client = get_acr_service_client(cli_ctx).registries
+    registry_list = client.list()
+
+    elements = [item for item in registry_list if item.name.lower() == registry_name.lower()]
+
+    if not elements:
+        return None
+    elif len(elements) > 1:
+        logger.warning("More than one registry object is found by the registry name.")
+        return None
+    return elements[0]
+
+
+def get_registry_by_login_server(cli_ctx, login_server):
```

Comment:
NIT: it seems that this method can be combined with `get_registry_by_registry_name` as they only differ in one line.